# Linkage mapping reveals loci that underlie differences in C. elegans growth
 
Growth rate and body size are integral and complex traits that contribute to the fitness of organisms. The identification of loci that underlie differences in these traits provide insights into the genetic contributors to development. Leveraging *Caenorhabditis elegans* as a tractable metazoan model for quantitative genetics, we can identify genetic regions that underlie differences in growth. We measured post-embryonic growth of the wild type strain (N2) and a natural isolate from Hawaii (CB4856), and found differences in body size. Using linkage mapping, we identified three distinct quantitative trait loci (QTL) on chromosome IV, V, and X that are associated with variation in body size. We further examined these size-associated QTL using chromosome substitution strains and near-isogenic lines, and validated the chromosome X QTL. Additionally, we generated a list of candidate genes identified in *C. elegans* for the chromosome X QTL. These genes could potentially contribute to differences in animal growth and should be evaluated in the future. Our work reveals the genetic architecture underlying animal growth and highlights the genetic complexity of body shape in the developing *C. elegans*. 

# Files
**S1 File. Raw animal growth data.** 
Raw growth data collected from the COPAS BIOSORT and processed using the easysorter R package to compile information from each well. (CSV)
**S2 File. Pruned animal growth data.**
Processed data from the COPAS BIOSORT following implementation of the mclust R package and removal of clusters containing non-animal objects. (CSV)
**S3 File. RIAIL phenotype data.** 
Residual phenotypic values for all 310 RIAILs, and parent strains (N2 and CB4856) for each trait (TOF = length and normEXT = width). (CSV)
**S4 File. Linkage mapping results.** 
Linkage mapping results with annotated QTL and confidence intervals. (CSV)
**S5 File. Summary of two-dimensional genome scan with length.** 
Summary of the scan2 object containing data from the two-dimensional genome scan with animal length (TOF). (CSV)
**S6 File. Summary of two-dimensional genome scan with width.**
Summary of the scan2 object containing data from the two-dimensional genome scan with animal width (normEXT). (CSV)
**S7 File. CSS/NIL sequence data.** 
VCF from the whole-genome sequencing for all the CSSs/NILs in this study. (DOCX)
**S8 File. CSS/NIL genotype data.** 
Simplified genotypes for the CSSs/NILs in this study. (CSV)
**S9 File. CSS and NIL raw phenotype data.** 
Raw phenotype data for the CSSs and NILs in this study. (CSV)
**S10 File. CSS and NIL pruned phenotype data.** 
Pruned phenotype data for the CSSs and NILs in this study. (CSV)
**S11 File. Statistical significance for mean animal length from CSS/NIL assay.** 
Pairwise statistical significance for all strains. (CSV)
**S12 File. Statistical significance for mean animal width from CSS/NIL assay.** 
Pairwise statistical significance for all strains. (CSV)
**S13 File. List of genes in the chromosome X QTL.** 
List of all genes in the chromosome X interval, their functional descriptions and GP annotations, and whether they have variation in the CB4856. (CSV)
**S1 Text. Reagents used to generate CSSs and NILs. (PDF)**
