---
title: "Supplemental Figures"
output: 
  html_document:
    toc: true
    toc_float: true
    theme: paper
    toc_depth: 4
---
```{r setup, echo=FALSE}
knitr::opts_chunk$set(
	echo = TRUE,
	message = FALSE,
	warning = FALSE
)
```

```{r load}
library(here)
library(tidyverse)
library(cowplot)
library(ggpubr)
library(ggridges)
library(sjPlot)

# S1 file
rawdata <- readr::read_csv(here::here("data", "S1File.csv")) %>%
  dplyr::mutate(strain = factor(strain, levels = c("N2","CB")))
pruneddata <- readr::read_csv(here::here("data", "S2File.csv")) %>%
  dplyr::mutate(strain = factor(strain, levels = c("N2","CB")))

nilpruneddata <- readr::read_csv(here::here("data", "S8File.csv"))

# S4 file
load(here::here("data","nilgeno.Rda"))
```

#### **S1 Fig. Raw measurements of animal size**

```{r S1Fig, fig.height=6, fig.width=10}
raw.length <- rawdata %>%
  dplyr::mutate(hour = as.numeric(timepoint)) %>%
  ggplot() +
  aes(x = hour, y = TOF) +
  geom_jitter(size = 0.01, width = 0.2, alpha = 0.02)+
  scale_x_continuous(breaks = seq(0, 60, 10)) +
  facet_wrap(~strain) +
  ylim(NA,750) +
  labs(x="Time (hours)", y = "Animal Length") +
  theme_cowplot(font_size = 10, rel_small = 8/10) +
  panel_border()

raw.width <- rawdata %>%
  dplyr::mutate(hour = as.numeric(timepoint)) %>%
  ggplot() +
  aes(x = hour, y = norm.EXT) +
  geom_jitter(size = 0.01, width = 0.2, alpha = 0.02)+
  scale_x_continuous(breaks = seq(0, 60, 10)) +
  facet_wrap(~strain) +
  panel_border() +
  labs(x="Time (hours)", y = "Animal Width") +
  theme_cowplot(font_size = 10, rel_small = 8/10) +
  panel_border()


figS1 <- cowplot::plot_grid(raw.length, raw.width, nrow = 2, ncol = 1, align = "hv", labels = "AUTO", label_size = 10)
figS1
```
```{r eval=FALSE, include=FALSE}
ggplot2::ggsave(here::here("figures", "supplement", "S1Fig.png"), plot = figS1, device = "png", width = 7, height = 5, units = "in", dpi = 300)
```


#### **S2 Fig. Gaussian finite mixture modeling of COPAS BIOSORT data**

```{r S2Fig, fig.height=5, fig.width=8}
removed <- dplyr::anti_join(rawdata, pruneddata) %>%
  dplyr::mutate(`Removed ?` = "Yes")

kept <- pruneddata %>%
  dplyr::mutate(`Removed ?` = "No")

total <- dplyr::full_join(removed, kept)

figS2 <- total %>%
  dplyr::filter(replicate == "R02" & timepoint %in% c("01","05",10,15,20,25,30,35,40,45,50)) %>%
  ggplot() +
  aes(x = log(TOF), y = log(EXT), color = `Removed ?`) +
  scale_color_manual(values = c("#BC3C29FF", "#0072B5FF")) +
  geom_jitter(size = 0.1, alpha = 0.5) +
  theme_cowplot(font_size = 10, rel_small = 8/10) +
  facet_wrap(~timepoint, nrow = 3) + panel_border()

figS2  
```
```{r eval=FALSE, include=FALSE}
ggplot2::ggsave(here::here("figures", "supplement", "S2Fig.png"), plot = figS2, device = "png", width = 8, height = 5, units = "in", dpi = 300)
```


#### **S3 Fig. Pruned measurements of animal size**
```{r S3Fig, fig.height=6, fig.width=10}
pruned.length <- pruneddata %>%
  dplyr::mutate(hour = as.numeric(timepoint)) %>%
  ggplot() +
  aes(x = hour, y = TOF) +
  geom_jitter(size = 0.01, width = 0.2, alpha = 0.02)+
  facet_wrap(~strain) +
  #ylim(NA,750) +
  scale_x_continuous(breaks = seq(0, 60, 5)) +
  labs(x="Time (hours)", y = "Animal Length") +
  theme_cowplot(font_size = 10, rel_small = 8/10) +
  panel_border()

pruned.width <- pruneddata %>%
  dplyr::mutate(hour = as.numeric(timepoint)) %>%
  ggplot() +
  aes(x = hour, y = norm.EXT) +
  geom_jitter(size = 0.01, width = 0.2, alpha = 0.02)+
  facet_wrap(~strain) +
  scale_x_continuous(breaks = seq(0, 70, 10)) +
  labs(x="Time (hours)", y = "Animal Width") +
  theme_cowplot(font_size = 10, rel_small = 8/10) +
  panel_border()

figS3 <- cowplot::plot_grid(pruned.length, pruned.width, nrow = 2, ncol = 1, align = "hv", labels = "AUTO", label_size = 10)
figS3
```
```{r eval=FALSE, include=FALSE}
ggplot2::ggsave(here::here("figures", "supplement", "S3Fig.png"), plot = figS3, device = "png", width = 7, height = 5, units = "in", dpi = 300)
```


#### **S4 Fig. Comparisons of means**
```{r Fig1, fig.height=6, fig.width=10}
sumdata <- pruneddata %>%
  dplyr::group_by(strain, timepoint, replicate, row, col) %>%
  dplyr::summarize(mean.Length = mean(TOF),
                   mean.Width = mean(norm.EXT), .groups = "drop")

ypos <- sumdata %>% 
  dplyr::group_by(timepoint) %>%
  dplyr::summarize(l = as.numeric(stats::quantile(mean.Length, probs = 0.75, na.rm = TRUE)[1]) + 40,
                   w = as.numeric(stats::quantile(mean.Width, probs = 0.50, na.rm = TRUE)[1]) + 0.35)

l <- sumdata %>%
  ggplot(., aes(x = timepoint, y = mean.Length, fill = strain)) +
  geom_boxplot(outlier.alpha = 0, alpha = 0.7) +
  geom_jitter(aes(color = strain), size=0.1, width=0.2, alpha=0.4) +
  theme_cowplot(font_size = 14, rel_small = 9/14) +
  scale_fill_manual(values = c("#ffa500","#0000ff")) +
  scale_color_manual(values = c("#ffa500","#0000ff")) +
  labs(x="Time (hours)", y = "Mean Animal Length") +
  ggpubr::stat_compare_means(method = "wilcox.test", label = "..p.signif..", label.y = ypos$l)

w <- sumdata %>%
  ggplot(., aes(x = timepoint, y = mean.Width, fill = strain)) +
  geom_boxplot(outlier.alpha = 0, alpha = 0.7) +
  geom_jitter(aes(color = strain), size=0.1, width=0.2, alpha=0.4) +
  theme_cowplot(font_size = 14, rel_small = 9/14) +
  scale_fill_manual(values = c("#ffa500","#0000ff")) +
  scale_color_manual(values = c("#ffa500","#0000ff")) +
  labs(x="Time (hours)", y = "Mean Animal Width") +
  ggpubr::stat_compare_means(method = "wilcox.test", label = "..p.signif..", label.y = ypos$w)

figS4 <- cowplot::plot_grid(l, w, labels = "AUTO", ncol = 1, nrow = 2, align = "hv", label_size = 10)
figS4
```
```{r eval=FALSE, include=FALSE}
ggsave(here::here("figures", "supplement", "S4-1Fig.png"), plot = figS4, device = "png", width = 12, height = 9, units = "in", dpi = 300, bg="white")
```

#### **S6 Fig. CSSs and NILs test length-associated chromosome IV QTL**
```{r S6Fig, fig.length=, fig.width=12, message=FALSE, warning=FALSE}
regress_stats <- function(phenodf, genodf, stages) {
  # regress out bleach effect
  regressed <- phenodf %>%
    dplyr::filter(stage %in% c(stages)) %>%
    dplyr::mutate(residual.meanTOF = residuals(aov(mean.TOF ~ bleach, data = .)),
                  residual.meanwidth = residuals(aov(mean.normEXT ~ bleach, data = .)))
  
  ### mean.TOF
  meanTOF.regressed <- regressed %>%
    summarize(comparison = row.names(TukeyHSD(aov(residual.meanTOF ~ strain))$strain),
              padj = TukeyHSD(aov(residual.meanTOF ~ strain))$strain[,4], .groups = "drop") 
  
  # ChrIV
  chrIV <- meanTOF.regressed %>%
    dplyr::filter(comparison %in% c("N2-ECA1064","N2-ECA597","CB4856-ECA575","CB4856-ECA599")) %>%
    tidyr::separate(comparison, into = c("Start", "End"), remove = FALSE) %>%
    dplyr::mutate(sig = dplyr::case_when(padj < 0.0001 ~ "****",
                                         padj < 0.001 ~ "***",
                                         padj < 0.01 ~ "**",
                                         padj < 0.05 ~ "*",
                                         TRUE ~ "ns"))
  # plot pheno
  chrIV.phenoplot <- regressed %>%
    dplyr::filter(strain %in% c("N2","CB4856","ECA1064","ECA597","ECA575","ECA599")) %>%
    ggplot(.) +
    aes(x = strain, y = residual.meanTOF, fill = strain) +
    geom_boxplot(outlier.color = NA, width = 0.6, alpha = 0.8, size = 0.2) +
    geom_jitter(width = 0.1, size = 0.2, alpha = 0.6) + 
    theme_cowplot() +
    scale_fill_manual(values = c("N2" = "#ffa500", "CB4856" = "#0000ff", "ECA575" = "gray","ECA1064" = "gray")) +
    facet_wrap(~stage, scales = "free") + 
    panel_border(color = "black", size = 0.8) +
    ggsignif::geom_signif(data = chrIV, manual = TRUE, inherit.aes = F,
                          aes(xmax = End, xmin = Start, y_position= c(60,45,60,45), annotations = sig)) +
    coord_flip() + facet_wrap(~"Phenotype") +
    scale_color_manual(values = c("N2" = "#ffa500", "CB4856" = "#0000ff")) + 
    guides(color = "none", size = "none", fill = "none") + ylab("Mean Animal Length") + xlab("") 
  
  # plot chrV genotype
  chrIV.genoplot <- genodf %>%
    dplyr::ungroup() %>%
    dplyr::filter(chrom == "IV",
                  sample %in% c("N2","ECA1064","ECA597","CB4856","ECA575","ECA599"),
                  !end == 18000000) %>%
    dplyr::mutate(chrom = paste0("chr", "IV"),
                  sample = factor(sample, levels = rev(c("N2","ECA1064","ECA597","CB4856","ECA575","ECA599")))) %>%
    dplyr::distinct() %>%
    ggplot(.)+
    geom_segment(aes(x = start/1e6, y = sample, xend = end/1e6, yend = sample, color = gt_name), size = 8) +
  facet_wrap(~chrom)+
    geom_vline(xintercept = 9.392639) +
    geom_vline(xintercept = c(6.211685,12.868784), linetype="dotted") +
    scale_color_manual(values=c("N2"="orange","CB4856"="blue"))+
    theme_cowplot() + panel_border(color = "black") +
    theme(axis.ticks.y = element_blank(),
          legend.position = "none",
          panel.grid.minor = element_blank(),
          panel.grid.major = element_blank())+
    labs(x = "Genomic position (Mb)", y = "") 
  
  # plot genome background
  chrIV.backplot <- genodf %>%
    dplyr::ungroup() %>%
    dplyr::filter(sample %in% c("N2","ECA1064","ECA597","CB4856","ECA575","ECA599"),
                  chrom == ifelse("IV" == "III", "II", "III")) %>%
    dplyr::mutate(bp = end - start) %>%
    dplyr::group_by(sample, gt_name) %>%
    dplyr::summarize(max = sum(bp)) %>%
    dplyr::arrange(desc(max)) %>%
    dplyr::filter(max == first(max)) %>%
    dplyr::mutate(start = 0,
                  end = 15e6,
                  chr = "Genome")  %>%
    dplyr::ungroup() %>%
    dplyr::mutate(sample = factor(sample, levels = rev(c("N2","ECA1064","ECA597","CB4856","ECA575","ECA599")))) %>%
    ggplot(.)+
    geom_segment(aes(x = start/1e6, y = sample, xend = end/1e6, yend = sample, color = gt_name), size = 8)+
    facet_wrap(~chr)+
    scale_color_manual(values=c("N2"="orange","CB4856"="blue"))+
    theme_cowplot() + panel_border(color = "black") +
    theme(axis.text = element_blank(),
          axis.title = element_blank(),
          legend.position = "none",
          axis.ticks = element_blank(),
          panel.grid = element_blank(),
          strip.text.y = element_blank())+
    labs(x = "Genomic position (Mb)", y = "")
  
  allIV <- cowplot::plot_grid(chrIV.genoplot, chrIV.backplot, chrIV.phenoplot,
                     nrow = 1, ncol = 3, rel_widths = c(3, 1, 4.5), align = "h", axis = "b", labels = c("A", "", "B"))
  
  return(allIV)
}

#summarize the pruned nil data
sumnils <- nilpruneddata %>%
  dplyr::mutate(strain = factor(strain, 
                                levels = rev(c("N2","ECA1064","ECA597","ECA2006","ECA232","ECA929",
                                               "CB4856","ECA575","ECA599","ECA1060","ECA1058","ECA828")))) %>%
  dplyr::group_by(stage,bleach,plate,strain,row,col) %>%
  dplyr::summarize(mean.TOF = mean(TOF),
                   mean.normEXT = mean(norm.EXT),
                   n = n(), .groups = "drop") 

figS6 <- regress_stats(sumnils, nilgeno, "L4s")
figS6
```
```{r eval=FALSE, include=FALSE}
ggplot2::ggsave(here::here("figures", "supplement", "S6Fig.png"), plot = figS6, device = "png", width = 12, height = 6, units = "in", dpi = 300, bg = "white")
```


#### **S5 Fig. CSSs and NILs test length-associated chromosome IV QTL**
```{r S5Fig, fig.length=, fig.width=12, message=FALSE, warning=FALSE}
regress_stats <- function(phenodf, genodf, stages) {
  # regress out bleach effect
  regressed <- phenodf %>%
    dplyr::filter(stage %in% c(stages)) %>%
    dplyr::mutate(residual.meanTOF = residuals(aov(mean.TOF ~ bleach, data = .)),
                  residual.meanwidth = residuals(aov(mean.normEXT ~ bleach, data = .)))
  
  ### mean.normEXT
  meanwidth.regressed <- regressed %>%
    summarize(comparison = row.names(TukeyHSD(aov(residual.meanwidth ~ strain))$strain),
              padj = TukeyHSD(aov(residual.meanwidth ~ strain))$strain[,4], .groups = "drop")
  
  # ChrV
  chrV <- meanwidth.regressed %>%
    dplyr::filter(comparison %in% c("N2-ECA2006","N2-ECA232","CB4856-ECA1060","CB4856-ECA1058")) %>%
    tidyr::separate(comparison, into = c("Start", "End"), remove = FALSE) %>%
    dplyr::mutate(sig = dplyr::case_when(padj < 0.0001 ~ "****",
                                         padj < 0.001 ~ "***",
                                         padj < 0.01 ~ "**",
                                         padj < 0.05 ~ "*",
                                         TRUE ~ "ns"))
  chrV.phenoplot <- regressed %>%
    dplyr::filter(strain %in% c("N2","CB4856","ECA2006","ECA232","ECA1060","ECA1058")) %>%
    ggplot(.) +
    aes(x = strain, y = residual.meanwidth, fill = strain) +
    geom_boxplot(outlier.color = NA, width = 0.6, alpha = 0.8, size = 0.2) +
    geom_jitter(width = 0.1, size = 0.2, alpha = 0.6) + 
    theme_cowplot() +
    scale_fill_manual(values = c("N2" = "#ffa500", "CB4856" = "#0000ff", "ECA1060" = "gray","ECA2006" = "gray")) +
    #facet_wrap(~stage, scales = "free") + 
    panel_border(color = "black", size = 0.8) +
    ggsignif::geom_signif(data = chrV, manual = TRUE, inherit.aes = F,
                          aes(xmax = End, xmin = Start, y_position= c(0.4,0.3,0.4,0.3), annotations = sig)) +
    coord_flip() + facet_wrap(~"Phenotype") +
    scale_color_manual(values = c("N2" = "#ffa500", "CB4856" = "#0000ff")) + 
    guides(color = "none", size = "none", fill = "none") + ylab("Mean Animal Width") + xlab("") 
  
  # plot chrV genotype
  chrV.genoplot <- genodf %>%
    dplyr::ungroup() %>%
    dplyr::filter(chrom == "V",
                  sample %in% c("N2","CB4856","ECA2006","ECA232","ECA1060","ECA1058"),
                  !end == 18000000) %>%
    dplyr::mutate(chrom = paste0("chr", "V"),
                  sample = factor(sample, levels = rev(c("N2","ECA2006","ECA232","CB4856","ECA1060","ECA1058")))) %>%
    dplyr::distinct() %>%
    ggplot(.)+
    geom_segment(aes(x = start/1e6, y = sample, xend = end/1e6, yend = sample, color = gt_name), size = 8) +
    facet_wrap(~chrom)+
    geom_vline(xintercept = 11.806498) +
    geom_vline(xintercept = c(5.371124,12.112105), linetype="dotted") +
    scale_color_manual(values=c("N2"="orange","CB4856"="blue"))+
    theme_cowplot() + panel_border(color = "black") +
    theme(axis.ticks.y = element_blank(),
          legend.position = "none",
          panel.grid.minor = element_blank(),
          panel.grid.major = element_blank())+
    labs(x = "Genomic position (Mb)", y = "") 
  
  # plot genome background
  chrV.backplot <- genodf %>%
    dplyr::ungroup() %>%
    dplyr::filter(sample %in% c("N2","ECA2006","ECA232","CB4856","ECA1060","ECA1058"),
                  chrom == ifelse("V" == "III", "II", "III")) %>%
    dplyr::mutate(bp = end - start) %>%
    dplyr::group_by(sample, gt_name) %>%
    dplyr::summarize(max = sum(bp)) %>%
    dplyr::arrange(desc(max)) %>%
    dplyr::filter(max == first(max)) %>%
    dplyr::mutate(start = 0,
                  end = 15e6,
                  chr = "Genome")  %>%
    dplyr::ungroup() %>%
    dplyr::mutate(sample = factor(sample, levels = rev(c("N2","ECA2006","ECA232","CB4856","ECA1060","ECA1058")))) %>%
    ggplot(.)+
    geom_segment(aes(x = start/1e6, y = sample, xend = end/1e6, yend = sample, color = gt_name), size = 8)+
    facet_wrap(~chr)+
    scale_color_manual(values=c("N2"="orange","CB4856"="blue"))+
    theme_cowplot() + panel_border(color = "black") +
    theme(axis.text = element_blank(),
          axis.title = element_blank(),
          legend.position = "none",
          axis.ticks = element_blank(),
          panel.grid = element_blank(),
          strip.text.y = element_blank())+
    labs(x = "Genomic position (Mb)", y = "")
  
  allV <- cowplot::plot_grid(chrV.genoplot, chrV.backplot, chrV.phenoplot,
                             nrow = 1, ncol = 3, rel_widths = c(3, 1, 4.5), 
                             align = "h", axis = "b", labels = c("A", "", "B"))
  
  return(allV)
}

figS7 <- regress_stats(sumnils, nilgeno, "L4s")
figS7
```
```{r eval=FALSE, include=FALSE}
ggplot2::ggsave(here::here("figures", "supplement", "S7Fig.png"), plot = figS7, device = "png", width = 12, height = 6, units = "in", dpi = 300, bg = "white")
```


#### **S7 and S8 Files**
```{r}
# regress out bleach effect
regressed <- sumnils %>%
    dplyr::filter(stage %in% c("L4s")) %>%
    dplyr::mutate(residual.meanTOF = residuals(aov(mean.TOF ~ bleach, data = .)),
                  residual.meanwidth = residuals(aov(mean.normEXT ~ bleach, data = .)))
```

Tukey's HSD: Mean animal length
```{r}
stats.length <- regressed %>%
  aov(residual.meanTOF ~ strain, data = .) %>%
  rstatix::tukey_hsd()
stats.length
```

Tukey's HSD: Mean animal width
```{r}
stats.width <- regressed %>%
  aov(residual.meanwidth ~ strain, data = .) %>%
  rstatix::tukey_hsd()
stats.width
```
















